This amazing database is a treasure trove of cinematic wonders from all around the globe.

Within this database, you'll discover two captivating tables. The first one, elegantly named 'directors,' showcases the names of renowned directors and their genders. The second table, titled 'movies,' is a mesmerizing collection of movie titles, along with intriguing details such as budgets, revenues, popularity scores, and more.

What makes this database truly remarkable is the wealth of insights it offers. You can uncover the finest cinematic gems by sorting movies based on their income, popularity, and votes. Additionally, you can delve into the realm of directorial excellence by determining the top director based on the total revenue generated by all of their movies.

In [15]:
# importing libraries
import numpy as np
import pandas as pd
import seaborn as sns
sns.set
import sqlite3

In [9]:
# Connecting to the existing SQLite database & 
# Creating a cursor object to execute SQL queries.
try:
    connection = sqlite3.connect(r'C:\Users\USER\Documents\Raw_Data\movies.sqlite')
    cursor = connection.cursor()
    print("Connected to the database")
except sqlite3.Error as e:
    print("Error connecting to the database:", e)
    exit(1)

Connected to the database


In [18]:
#getting column names for movies table
cursor.execute("PRAGMA table_info(movies)")
table_info = cursor.fetchall()

In [19]:
#get table names
movies_column_names = []
for column_info in table_info:
    column_name = column_info[1]
    movies_column_names.append(column_name)
movies_column_names

['id',
 'original_title',
 'budget',
 'popularity',
 'release_date',
 'revenue',
 'title',
 'vote_average',
 'vote_count',
 'overview',
 'tagline',
 'uid',
 'director_id']

In [11]:
# Using the column name to print the movie dataframe
cursor.execute('SELECT * FROM movies')
movies = cursor.fetchall()
movies = pd.DataFrame(movies, columns = ['id', 'original_title', 'budget', 'popularity', 'release_date', 'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagline', 'uid', 'director_id'])
movies.head(3)

id                            original_title     budget  popularity  \
0  43597                                    Avatar  237000000         150   
1  43598  Pirates of the Caribbean: At World's End  300000000         139   
2  43599                                   Spectre  245000000         107   

  release_date     revenue                                     title  \
0   2009-12-10  2787965087                                    Avatar   
1   2007-05-19   961000000  Pirates of the Caribbean: At World's End   
2   2015-10-26   880674609                                   Spectre   

   vote_average  vote_count  \
0           7.2       11800   
1           6.9        4500   
2           6.3        4466   

                                            overview  \
0  In the 22nd century, a paraplegic Marine is di...   
1  Captain Barbossa, long believed to be dead, ha...   
2  A cryptic message from Bond’s past sends him o...   

                                          tagline     uid  director_id  
0                     Enter the World of Pandora.   19995         4762  
1  At the end of the world, the adventure begins.     285         4763  
2                           A Plan No One Escapes  206647         4764

In [20]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4773 entries, 0 to 4772
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              4773 non-null   int64  
 1   original_title  4773 non-null   object 
 2   budget          4773 non-null   int64  
 3   popularity      4773 non-null   int64  
 4   release_date    4773 non-null   object 
 5   revenue         4773 non-null   int64  
 6   title           4773 non-null   object 
 7   vote_average    4773 non-null   float64
 8   vote_count      4773 non-null   int64  
 9   overview        4770 non-null   object 
 10  tagline         3951 non-null   object 
 11  uid             4773 non-null   int64  
 12  director_id     4773 non-null   int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 484.9+ KB


2. How can one obtain a comprehensive dataset on directors?

In [25]:
#getting column names for directors table
cursor.execute("PRAGMA table_info(directors)")
table_info = cursor.fetchall()

In [26]:
#get table names
movies_column_names = []
for column_info in table_info:
    column_name = column_info[1]
    movies_column_names.append(column_name)
movies_column_names

['name', 'id', 'gender', 'uid', 'department']

In [28]:
# Using the column name to print the directors dataframe
cursor.execute('SELECT * FROM directors')
directors = cursor.fetchall()
directors = pd.DataFrame(directors, columns = ['name', 'id', 'gender', 'uid', 'department'])
directors.head()

name    id  gender   uid department
0      James Cameron  4762       2  2710  Directing
1     Gore Verbinski  4763       2  1704  Directing
2         Sam Mendes  4764       2    39  Directing
3  Christopher Nolan  4765       2   525  Directing
4     Andrew Stanton  4766       2     7  Directing

In [29]:
directors.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2349 entries, 0 to 2348
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   name        2349 non-null   object
 1   id          2349 non-null   int64 
 2   gender      2349 non-null   int64 
 3   uid         2349 non-null   int64 
 4   department  2349 non-null   object
dtypes: int64(3), object(2)
memory usage: 91.9+ KB


# Insights from the database

In [42]:
# number of movies in the datase
cursor.execute('SELECT COUNT(*) FROM movies')
movies_count = cursor.fetchall()
print("The iMDB database contains information about "+str(movies_count[0][0])+" movies.")

The iMDB database contains information about 4773 movies.


In [43]:
# movies with highest production cost
cursor.execute('SELECT original_title, name, release_date, budget, revenue FROM movies JOIN directors ON movies.director_id = directors.id ORDER BY budget DESC')
budget = cursor.fetchall()
budget = pd.DataFrame(me, columns = ['original_title', 'director_name', 'release_date', 'budget', 'revenue'])
budget.head(5)

original_title   director_name release_date  \
0  Pirates of the Caribbean: On Stranger Tides    Rob Marshall   2011-05-14   
1     Pirates of the Caribbean: At World's End  Gore Verbinski   2007-05-19   
2                      Avengers: Age of Ultron     Joss Whedon   2015-04-22   
3                             Superman Returns    Bryan Singer   2006-06-28   
4                                  John Carter  Andrew Stanton   2012-03-07   

      budget     revenue  
0  380000000  1045713802  
1  300000000   961000000  
2  280000000  1405403694  
3  270000000   391081192  
4  260000000   284139100

> ## Pirates of the Caribbean: On Stranger Tides
Directed By Rob Marshall

Is The movie with the highest Production Cost, USD 380,000,000.
Followed by Pirates of the Cribbean: At World's End, then Angers: Age of Ultron

In [46]:
# Popularity
cursor.execute('SELECT original_title, name, release_date, popularity, revenue FROM movies JOIN directors ON movies.director_id = directors.id ORDER BY popularity DESC')
most_Popular = cursor.fetchall()
most_Popular = pd.DataFrame(most_Popular, columns = ['original_title', 'director_name', 'release_date', 'popularity', 'revenue'])
most_Popular.head(5)

original_title      director_name release_date  popularity  \
0                  Minions         Kyle Balda   2015-06-17         875   
1             Interstellar  Christopher Nolan   2014-11-05         724   
2                 Deadpool         Tim Miller   2016-02-09         514   
3  Guardians of the Galaxy         James Gunn   2014-07-30         481   
4       Mad Max: Fury Road      George Miller   2015-05-13         434   

      revenue  
0  1156730962  
1   675120017  
2   783112979  
3   773328629  
4   378858340

> ## Minions :
Directed by Kyle Balda

Is the most popular movie on the list with 875 points
Followed by Interstellar, and them Deadpool on the third spot

In [48]:
# Revenue
cursor.execute('SELECT original_title, name, release_date, revenue FROM movies JOIN directors ON movies.director_id = directors.id ORDER BY revenue DESC')
revenue = cursor.fetchall()
revenue = pd.DataFrame(revenue, columns = ['original_title', 'director_name', 'release_date', 'revenue'])
revenue.head(5)

original_title    director_name release_date     revenue
0          Avatar    James Cameron   2009-12-10  2787965087
1         Titanic    James Cameron   1997-11-18  1845034188
2    The Avengers      Joss Whedon   2012-04-25  1519557910
3  Jurassic World  Colin Trevorrow   2015-06-09  1513528810
4       Furious 7        James Wan   2015-04-01  1506249360

> ## Avatar :
Directed by James Cameron


Has the highest revenue of $2,787,965,087. 
Followed By Titanic, The Avengers making the third on the table.

In [74]:
# number of female Directors
cursor.execute('SELECT COUNT(*) FROM directors WHERE gender==1')
dir_female = cursor.fetchall()
print("Number of female directors on the list: "+str(dir_female[0][0]))

Number of female directors on the list: 150


In [75]:
cursor.execute('SELECT original_title, release_date, revenue, popularity, vote_average FROM directors JOIN movies ON directors.id = movies.director_id WHERE name = "Steven Spielberg" ORDER BY release_date DESC')
Steven_Spielberg = cursor.fetchall()
Steven_Spielberg = pd.DataFrame(Steven_Spielberg, columns = ['original_title', 'release_date', 'revenue', 'popularity', 'vote_average'])
Steven_Spielberg

original_title release_date    revenue  \
0                                             The BFG   2016-06-01  183345589   
1                                     Bridge of Spies   2015-10-15  165478348   
2                                             Lincoln   2012-11-09  275293450   
3                                           War Horse   2011-12-25  177584879   
4                            The Adventures of Tintin   2011-10-25  371940071   
5   Indiana Jones and the Kingdom of the Crystal S...   2008-05-21  786636033   
6                                              Munich   2005-12-22  130358911   
7                                   War of the Worlds   2005-06-28  591739379   
8                                        The Terminal   2004-06-17  219417255   
9                                 Catch Me If You Can   2002-12-25  352114312   
10                                    Minority Report   2002-06-20  358372926   
11                       A.I. Artificial Intelligence   2001-06-29  235926552   
12                                Saving Private Ryan   1998-07-24  481840909   
13                                            Amistad   1997-12-03   74000000   
14                      The Lost World: Jurassic Park   1997-05-23  229074524   
15                                   Schindler's List   1993-11-29  321365567   
16                                      Jurassic Park   1993-06-11  920100000   
17                                               Hook   1991-12-11  300854823   
18                 Indiana Jones and the Last Crusade   1989-05-24  474171806   
19                                   The Color Purple   1985-12-18  146292009   
20               Indiana Jones and the Temple of Doom   1984-05-23  333000000   
21                           Twilight Zone: The Movie   1983-06-24   29450919   
22                         E.T. the Extra-Terrestrial   1982-04-03  792910554   
23                            Raiders of the Lost Ark   1981-06-12  389925971   
24                                               1941   1979-12-13   31755742   
25                 Close Encounters of the Third Kind   1977-11-16  303788635   
26                                               Jaws   1975-06-18  470654000   

    popularity  vote_average  
0           44           6.0  
1           48           7.2  
2           36           6.7  
3           29           7.0  
4           89           6.7  
5           75           5.7  
6           29           6.9  
7           48           6.2  
8           57           7.0  
9           73           7.7  
10          65           7.1  
11          34           6.8  
12          76           7.9  
13           3           6.8  
14           2           6.2  
15         104           8.3  
16          40           7.6  
17          33           6.6  
18          80           7.6  
19          17           7.7  
20          66           7.1  
21          12           6.2  
22          56           7.3  
23          68           7.7  
24          10           5.6  
25          52           7.2  
26          50           7.5

> ## Steven Spielber
It Can Be Understood That Steven Spielberg Has Produced Many And Incredible Movies That Are Shows In The Table Above.

In [57]:
# james Cameron Movies
cursor.execute('SELECT original_title, release_date, budget, revenue, popularity, vote_average FROM directors JOIN movies ON directors.id = movies.director_id WHERE name = "James Cameron" ORDER BY revenue DESC')
JamesCameron = cursor.fetchall()
JamesCameron = pd.DataFrame(JamesCameron, columns = ['original_title', 'release_date', 'budget', 'revenue', 'popularity', 'vote_average'])
JamesCameron

original_title release_date     budget     revenue  popularity  \
0                      Avatar   2009-12-10  237000000  2787965087         150   
1                     Titanic   1997-11-18  200000000  1845034188         100   
2  Terminator 2: Judgment Day   1991-07-01  100000000   520000000         101   
3                   True Lies   1994-07-14  115000000   378882411          38   
4                      Aliens   1986-07-18   18500000   183316455          67   
5                   The Abyss   1989-08-09   70000000    90000098          24   
6              The Terminator   1984-10-26    6400000    78371200          74   

   vote_average  
0           7.2  
1           7.5  
2           7.7  
3           6.8  
4           7.7  
5           7.1  
6           7.3

> ## James Cmeron : 
Has seven movies in the database, list as seen above

In [60]:
# Movie Rating
cursor.execute('SELECT original_title, name, release_date, vote_average, vote_count FROM movies JOIN directors ON movies.director_id = directors.id ORDER BY vote_average DESC')
voting = cursor.fetchall()
voting = pd.DataFrame(voting, columns = ['original_title', 'director_name', 'release_date',  'vote_average', 'vote_count'])
voting.head(5)

original_title   director_name release_date  vote_average  \
0          Stiff Upper Lips     Gary Sinyor   1998-06-12          10.0   
1     Dancer, Texas Pop. 81   Tim McCanlies   1998-05-01          10.0   
2                 Sardaarji    Rohit Jugraj   2015-06-26           9.5   
3            One Man's Hero      Lance Hool   1999-08-02           9.3   
4  The Shawshank Redemption  Frank Darabont   1994-09-23           8.5   

   vote_count  
0           1  
1           1  
2           2  
3           2  
4        8205

> ## Stiff Upper Lips :
Directed by Gary Sinyo
> ## Dancer, Texas Pop. 81:
Directed by Tim McCanlies


The two movies above have the highest voting average of 10 each, but it turns out that they both got voted by one person each which allows them to have 100% voting averages

In [62]:
# Directors' Revenue
cursor.execute('SELECT name, COUNT(original_title), SUM(revenue)FROM directors JOIN movies ON directors.id = movies.director_id GROUP BY name ORDER BY SUM(revenue) DESC')
Director_rev = cursor.fetchall()
Director_rev = pd.DataFrame(Director_rev, columns = ['director_name', 'movies', 'revenue'])
Director_rev.head(5)

director_name  movies     revenue
0   Steven Spielberg      27  9147393164
1      Peter Jackson       9  6498642820
2      James Cameron       7  5883569439
3        Michael Bay      12  5832524638
4  Christopher Nolan       8  4227483234

> ## Steven Spielberg :


Is The Director With The Most Gross Profit, with USD 9,147,393,164 from his 27 movies. 
Then, Peter Jackson With USD 6,498,642,820, James Cameron With USD 5,883,569,439, And The Other Directors.

In [67]:
cursor.execute('SELECT name, COUNT(original_title), SUM(revenue) FROM directors JOIN movies ON directors.id = movies.director_id GROUP BY name ORDER BY COUNT(original_title) DESC')
Director_pro = cursor.fetchall()
Director_pro = pd.DataFrame(Director_pro, columns = ['director_name', 'movies', 'revenue'])
Director_pro.head(5)

director_name  movies     revenue
0  Steven Spielberg      27  9147393164
1       Woody Allen      21   669101038
2   Martin Scorsese      20  1956635998
3    Clint Eastwood      20  2512058888
4         Spike Lee      16   340618771


> ## Steven Spielberg :

Steven Spielberg Is The Most Productive Director With 27 Movies he Has Directed, followed by Woody Allen Has Only 21 Movies Titles, and then Martin Scorsese And Clint Eastwood With 20 Movies Titles each.
